sdk/core/System.ClientModel/src/Message/PipelineMessage.cs

```diff
@@ -117,26 +132,24 @@ public void Apply(RequestOptions options)
 
     #endregion
 
-    internal void AssertResponse()
+    #region IDisposable
+
+    public void Dispose()
     {
-        if (Response is null)
-        {
-            throw new InvalidOperationException("Response is not set on message.");
-        }
+        Dispose(true);
+        GC.SuppressFinalize(this);
     }
 
-    #region IDisposable
-
     protected virtual void Dispose(bool disposing)
     {
         if (disposing && !_disposed)
         {
-            var request = Request;
+            PipelineRequest request = Request;
             request?.Dispose();
 
             _propertyBag.Dispose();
 
-            var response = Response;
+            PipelineResponse? response = Response;
             if (response != null)
```

Comment:
should this use the same ? syntax as the above?

Reply:
Yep!